ql/src/java/org/apache/hadoop/hive/ql/DriverContext.java

```diff
@@ -137,13 +152,26 @@ public void setPlan(QueryPlan plan) {
       this.runtimeContext = Utilities.getFirstTezTask(plan.getRootTasks())
           .map(TezTask::getRuntimeContext)
           .orElse(null);
+      this.queryProperties = plan.getQueryProperties();
     }
   }
 
+  public String getQueryType() {
+    return queryType == null ? "UNKNOWN" : queryType;
```

Comment:
why `null` handling is different from the `getDdlType`

Reply:
a query must have a type under ideal circumstances, and if we failed to figure out, that's where "UNKNOWN" comes into the picture, whereas ddlType is not applicable at all for non-DDL queries, does this make sense? do you think it's better to queryType=="" as the dummy value?

Reply:
empty string should be ok